hadoop-tools/hadoop-azure/src/test/java/org/apache/hadoop/fs/azure/MockStorageInterface.java

```diff
@@ -137,9 +139,19 @@ private static String convertUriToDecodedString(URI uri) {
 
   private static URI convertKeyToEncodedUri(String key) {
     try {
-      return new URIBuilder().setPath(key).build();
+      URI unEncodedURI = new URI(key);
+      return new URIBuilder().setPath(unEncodedURI.getRawPath())
+          .setScheme(unEncodedURI.getScheme()).build();
```

Comment:
Seems only one test is failing, We can first convert the `key` to `Path` and then use `getPath` instead of `getRawPath` in `URIBuilder`. Something like this worked for me -
 ```suggestion
      Path p = new Path(key);
      URI unEncodedURI = p.toUri();
      return new URIBuilder().setPath(unEncodedURI.getPath())
          .setScheme(unEncodedURI.getScheme()).build();
```
Give a try, if this works, if everything comes out to be safe post your testing.